src/main/java/ru/mercuriev/game/of/life/graph/GraphEngine.java

```diff
@@ -0,0 +1,33 @@
+package ru.mercuriev.game.of.life.graph;
+
+import ru.mercuriev.game.of.life.graph.cell.Cell;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Engine interface implementation
+ * Uses Cell uder the hood to calculate next generation
+ *
+ * @author paul
+ */
+// TODO implement Engine interface
+public class GraphEngine {
+
+    // TODO @Override
+    public int[][] next(int[][] currentGeneration) {
+
+        // int[][] -> Stream<IntStream>
+        Stream<IntStream> input = Arrays.stream(currentGeneration)
```

Comment:
What about ```Stream<Stream<Integer>>```? I think it's more readable. When I saw this code at first time I was confused and think that you flattalize input with your custom class ```IntStream``` in it's basis.